tools/cxbe/Xbe.cpp

```diff
@@ -376,14 +376,27 @@ Xbe::Xbe(class Exe *x_Exe, const char *x_szTitle, uint32 x_dwTitleID, uint32 x_d
 
                 memset(&m_SectionHeader[v].dwFlags, 0, sizeof(m_SectionHeader->dwFlags));
 
-                if(characteristics & IMAGE_SCN_MEM_WRITE)
-                    m_SectionHeader[v].dwFlags.bWritable = true;
+                // check for $$XTIMAGE or $$XSIMAGE and set the correct flags
+                if(x_Exe->m_SectionHeader_longname[v].m_longname &&
+                   (!strcmp(x_Exe->m_SectionHeader_longname[v].m_longname, "$$XTIMAGE") ||
+                    !strcmp(x_Exe->m_SectionHeader_longname[v].m_longname, "$$XSIMAGE")))
```

Comment:
I'm not a fan of this hack. I'd rather like to see proper file insertion support in cxbe and something like `-TITLEIMAGE:` / `-SAVEIMAGE:` or similar.

Reply:
There is currently no code to do section insertion as far as i'm aware, and that could probably be removed later once there is code made to add sections.

Reply:
There isn't, as I said I'd rather see it implemented than a name-based hack on top of violating the PE spec. nxdk does not require the usage of these sections nor does it implement the APIs relying on them yet, so we might as well take the time to do things properly.

Reply:
I don't have the time to write something like that and I need the feature now. Loading the image sections takes up ram to do nothing. This code can be removed later when the better code is added.

Reply:
> This code can be removed later when the better code is added.
I'm not really interested in nxdk acquiring *even more* technical debt. We haven't even gotten rid of the one inherited from OpenXDK yet.